### PR TITLE
fix(sdk): mul gas by 1.5

### PR DESCRIPTION
**Pull request by roninjin10**

### CODE REVIEW

**Comment by roninjin10** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #6045** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6045" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6045?utm_source=stack-comment).

File: packages/sdk/src/cross-chain-messenger.ts [hunk 1]
```diff
@@ -1870,11 +1870,20 @@ export class CrossChainMessenger {
         overrides?: PayableOverrides
       }
     ): Promise<TransactionRequest> => {
+      const gasEstimation = await this.estimateGas.depositETH(amount, opts)
```
Comment (mslipper) [inline packages/sdk/src/cross-chain-messenger.ts:1873]: Do we need to do the same thing for ERC20s?

File: packages/sdk/src/cross-chain-messenger.ts [hunk 2]
```diff
@@ -1947,7 +1956,21 @@ export class CrossChainMessenger {
       }
     ): Promise<TransactionRequest> => {
       const bridge = await this.getBridgeForTokenPair(l1Token, l2Token)
-      return bridge.populateTransaction.deposit(l1Token, l2Token, amount, opts)
+      const gasEstimation = await this.estimateGas.depositETH(amount, opts)
```
Comment (mslipper) [inline packages/sdk/src/cross-chain-messenger.ts:1959]: `deposit` rather than `depositETH`?

File: packages/sdk/src/cross-chain-messenger.ts [hunk 3]
```diff
@@ -1868,13 +1868,28 @@ export class CrossChainMessenger {
         recipient?: AddressLike
         l2GasLimit?: NumberLike
         overrides?: PayableOverrides
-      }
+      },
+      isEstimatingGas: boolean = false
```
Comment (roninjin10) [inline packages/sdk/src/cross-chain-messenger.ts:1872]: infinite loop if we don't do this

File: packages/sdk/src/cross-chain-messenger.ts [hunk 4]
```diff
@@ -1868,13 +1868,28 @@ export class CrossChainMessenger {
...
     ): Promise<TransactionRequest> => {
+      const getOpts = async () => {
+        if (isEstimatingGas) {
+          return opts
+        }
+        const gasEstimation = await this.estimateGas.depositETH(amount, opts)
+        gasEstimation.mul(1.5)
+        return {
+          ...opts,
```
Comment (roninjin10) [inline packages/sdk/src/cross-chain-messenger.ts:1881]: if opts is udnefined this doesn't blow up

File: packages/sdk/src/cross-chain-messenger.ts [hunk 5]
```diff
@@ -1868,13 +1868,28 @@ export class CrossChainMessenger {
...
-      }
+      },
+      isEstimatingGas: boolean = false
     ): Promise<TransactionRequest> => {
+      const getOpts = async () => {
+        if (isEstimatingGas) {
+          return opts
+        }
+        const gasEstimation = await this.estimateGas.depositETH(amount, opts)
+        gasEstimation.mul(1.5)
```
Comment (tynes) [inline packages/sdk/src/cross-chain-messenger.ts:1879]: you need to assign here to `gasEstimation`, it doesnt mutate, it copies

Comment (roninjin10) [reply packages/sdk/src/cross-chain-messenger.ts:1879]: holy my brain isn't working wtf how'd I read this without noticing it?

Comment (roninjin10) [reply packages/sdk/src/cross-chain-messenger.ts:1879]: Oh I see actually this is right I just have a line of code I didn't delete.  I do this on the next line this line of code is just dead code

**Comment by roninjin10** [comment]
Squashed to one commit waiting for tests the pass then this is good

**Review by mslipper — APPROVED** [review]

**Pull request merged**